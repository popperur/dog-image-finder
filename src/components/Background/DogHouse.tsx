import {
  DogHouseContainer,
  RoofLeft,
  RoofRight,
  RoofRightWall,
  WallLeft,
  WallRight,
  WallRightDoor,
} from "components/Background/styles.tsx";

function DogHouse() {
  return (
    <DogHouseContainer>
      <RoofRight>
        <RoofRightWall />
      </RoofRight>
      <WallLeft />
      <WallRight>
        <WallRightDoor />
      </WallRight>
      <RoofLeft />
    </DogHouseContainer>
  );
}

export default DogHouse;
