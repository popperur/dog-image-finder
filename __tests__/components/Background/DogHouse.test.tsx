import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import DogHouse from "components/Background/DogHouse";

describe("DogHouse component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<DogHouse />);

    expect(asFragment()).toMatchSnapshot();
  });
});
