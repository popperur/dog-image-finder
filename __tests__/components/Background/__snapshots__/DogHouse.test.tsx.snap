// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DogHouse component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  bottom: 0;
  left: 27%;
  width: 344px;
  height: 271px;
  z-index: 90;
}

.c6 {
  position: absolute;
  top: 0;
  left: -16px;
  width: 253px;
  height: 141px;
  background-color: #1e4857;
  transform: skew(-33deg);
  border-right: 8px solid #4d4a55;
  border-bottom: 8px solid #4d4a55;
  border-bottom-right-radius: 5px;
  border-bottom-left-radius: 5px;
}

.c1 {
  position: absolute;
  top: 0;
  left: 37px;
  width: 292px;
  height: 140px;
  background-color: #1a3140;
  transform: skew(30deg);
  border-top-left-radius: 100%;
  border-top-right-radius: 12px;
  border-right: 8px solid #4d4a55;
  border-top: 8px solid #4d4a55;
}

.c2 {
  position: absolute;
  top: 0;
  left: 32px;
  width: 240px;
  height: 140px;
  background-color: #8c2c1f;
  transform: skew(0deg);
  border-bottom-right-radius: 16px;
}

.c3 {
  position: absolute;
  bottom: 0;
  width: 180px;
  height: 83%;
  background-color: #622224;
  border-top-left-radius: 5px;
}

.c4 {
  position: absolute;
  bottom: 0;
  right: -1px;
  width: 165px;
  height: 49%;
  background-color: #8c2c1f;
}

.c5 {
  position: absolute;
  top: 6%;
  left: 25px;
  height: 81%;
  width: 70%;
  background-color: #642220;
  border-radius: 50%;
  box-shadow: -14px 0 0 #5e2c2d inset;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    />
    <div
      class="c4"
    >
      <div
        class="c5"
      />
    </div>
    <div
      class="c6"
    />
  </div>
</DocumentFragment>
`;
